metadata-io/src/main/java/com/linkedin/metadata/service/GlossaryTermService.java

```diff
@@ -0,0 +1,469 @@
+    if (!glossaryTerms.hasTerms()) {
+      glossaryTerms.setTerms(new GlossaryTermAssociationArray());
+    }
+
+    GlossaryTermAssociationArray glossaryTermAssociationArray = glossaryTerms.getTerms();
+
+    List<Urn> glossaryTermsToAdd = new ArrayList<>();
+    for (Urn glossaryTermUrn : glossaryTermUrns) {
+      if (glossaryTermAssociationArray.stream().anyMatch(association -> association.getUrn().equals(glossaryTermUrn))) {
+        continue;
+      }
+      glossaryTermsToAdd.add(glossaryTermUrn);
+    }
+
+    // Check for no glossaryTerms to add
+    if (glossaryTermsToAdd.size() == 0) {
+      return;
+    }
+
+    for (Urn glossaryTermUrn : glossaryTermsToAdd) {
+      GlossaryTermAssociation newAssociation = new GlossaryTermAssociation();
+      newAssociation.setUrn(GlossaryTermUrn.createFromUrn(glossaryTermUrn));
+      glossaryTermAssociationArray.add(newAssociation);
+    }
+  }
+
+  private static GlossaryTermAssociationArray removeGlossaryTermsIfExists(GlossaryTerms glossaryTerms, List<Urn> glossaryTermUrns) {
+    if (!glossaryTerms.hasTerms()) {
+      glossaryTerms.setTerms(new GlossaryTermAssociationArray());
+    }
+    GlossaryTermAssociationArray glossaryTermAssociationArray = glossaryTerms.getTerms();
+    for (Urn glossaryTermUrn : glossaryTermUrns) {
+      glossaryTermAssociationArray.removeIf(association -> association.getUrn().equals(glossaryTermUrn));
+    }
+    return glossaryTermAssociationArray;
+  }
+
+  @Nonnull
+  private Map<Urn, GlossaryTerms> getGlossaryTermsAspects(
+      @Nonnull Set<Urn> entityUrns,
+      @Nonnull GlossaryTerms defaultValue,
+      @Nonnull Authentication authentication) {
+
+    if (entityUrns.size() <= 0) {
+      return Collections.emptyMap();
+    }
+
+    try {
+      Map<Urn, Aspect> aspects = batchGetLatestAspect(
+          entityUrns.stream().findFirst().get().getEntityType(), // TODO Improve this.
+          entityUrns,
+          Constants.GLOSSARY_TERMS_ASPECT_NAME,
+          this.entityClient,
+          authentication
+      );
+
+      final Map<Urn, GlossaryTerms> finalResult = new HashMap<>();
+      for (Urn entity : entityUrns) {
+        Aspect aspect = aspects.get(entity);
+        if (aspect == null) {
+          finalResult.put(entity, defaultValue);
+        } else {
+          finalResult.put(entity, new GlossaryTerms(aspect.data()));
+        }
+      }
+      return finalResult;
+    } catch (Exception e) {
+      log.error(
+          "Error retrieving glossary terms for entities. Entities: {} aspect: {}",
+          entityUrns,
+          Constants.GLOSSARY_TERMS_ASPECT_NAME,
+          e);
+      return Collections.emptyMap();
+    }
+  }
+
+  @Nonnull
+  private Map<Urn, EditableSchemaMetadata> getEditableSchemaMetadataAspects(
+      @Nonnull Set<Urn> entityUrns,
+      @Nonnull EditableSchemaMetadata defaultValue,
+      @Nonnull Authentication authentication) {
+
+    if (entityUrns.size() <= 0) {
+      return Collections.emptyMap();
+    }
+
```

Comment:
copy pasted method? base class / abstract class would resolve this. or a util method.